src/main.rs

```diff
@@ -0,0 +1,204 @@
+#[macro_use]
+extern crate serde;
+extern crate tokio;
+
+use chrono::DateTime;
+use chrono::Local;
+use chrono::SubsecRound;
+use chrono::TimeDelta;
+use serde_json::Deserializer;
+use serde_json::Value;
+use std::borrow::Borrow;
+use std::collections::HashMap;
+use tokio::task::JoinSet;
+use tokio::time::sleep;
+
+#[derive(Debug, Clone)]
+pub enum Instruction {
+    Lighting(Lights),
+    Pumping(Pumps),
+}
+
+#[derive(Debug, Clone)]
+pub enum Lights {
+    LightsOn(DateTime<Local>),
+    LightsOff(DateTime<Local>),
+}
+
+#[derive(Debug, Clone)]
+pub enum Pumps {
+    PumpsOn(DateTime<Local>),
+    PumpsOff(DateTime<Local>),
+}
+
+impl Instruction {
+    pub fn inspect(self) -> DateTime<Local> {
+        match self {
+            Instruction::Pumping(Pumps::PumpsOn(x)) => x,
+            Instruction::Pumping(Pumps::PumpsOff(x)) => x,
+            Instruction::Lighting(Lights::LightsOn(x)) => x,
+            Instruction::Lighting(Lights::LightsOff(x)) => x,
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() -> () {
+    let current_datetime = chrono::Local::now();
+
+    let demo_json_contents: Value = serde_json::from_str(
+        &(tokio::fs::read_to_string("demo.json")
+            .await
+            .expect("reading in JSON to work")),
+    )
+    .expect("JSON Value to work after read-in");
+    println!(
+        "JSON is working inside main, here are some untyped values,  {:?}, cool? {:?}",
+        demo_json_contents["devices"], demo_json_contents["cool"]
+    );
+
+    // TODO JSON file for schedule (MVP)
+    let schedule_json_contents: Value = serde_json::from_str(
+        &(tokio::fs::read_to_string("schedule.json")
+            .await
+            .expect("reading in JSON to work")),
+    )
+    .expect("JSON Value to work after read-in");
+    println!(
+        "JSON schedule is working inside main, {:?}",
+        schedule_json_contents["schedule"]
+    );
+    /*let sched: <_> = serde_json::from_value(schedule_json_contents)
+            .expect("JSON schedule contents to provide Value");
+        println!("{:?}", sched);
+    */
+
+    // command schedule is hard coded from file, so not good for demo
+    let mut _command_schedule: Vec<Instruction> = Vec::new();
+
+    // for demonstration, we will hard code a single day demo schedule
+    // multi day schedules could be generated algorithmically, ie, same for 12 days or, reduce light by 5/min a day for 40 days, etc
+    // would this be good as a test?
```

Comment:
Cool, I will plan to pull out some of this 'demo schedule' code to a stashed file, to later build tests from.
So a lot of this will disappear from the PR.